File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let minPrice = Number.MAX_SAFE_INTEGER;
    let maxProfite = 0;
    for(let price of prices){
        minPrice = Math.min(minPrice,price);
        maxProfite = Math.max(maxProfite,price - minPrice );
    }
    return maxProfite;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;
        for(int price : prices){
            minPrice = Math.min(minPrice,price);
            int profit = price - minPrice;
            maxProfit = Math.max(profit,maxProfit); 
        }
        return maxProfit;
    }
}